### ISSUE TITLE
PAL_VirtualUnwindOutOfProc isn't implemented for non x64 platforms

### ISSUE THREAD

**Issue by chsienki** [dotnet/coreclr#15062]
On platforms other than x64 _PAL_VirtualUnwindOutOfProc(...)_ is stubbed out to return FALSE  ([remote-unwind.cpp#L1081](https://github.com/dotnet/coreclr/blob/88527bb6bdec8662a5cef5efcadb22c7fcabae50/src/pal/src/exception/remote-unwind.cpp#L1081)).

The current implementation falls back to using DataTarget4->VirtualUnwind on platforms where the PAL method is not implemented, but this will not work in the dump debugging scenario.

**Comment by chsienki** [comment]
cc @mikem8361. I don't have permission to assign this to you.

**Comment by mikem8361** [comment]
I assigned to myself.

**Comment by lt72** [comment]
@mikem8361 what scenarios do we lose if we do not server this for 2.1?

**Comment by mikem8361** [comment]
If production bps are supported on arm32/arm64 platforms, this could be a problem.  Linux arm32/arm64 core dumps will be a problem but VS/VSCode doesn't support managed core dumps yet so that isn't a big deal. Any place we want ICLRDebugging::OpenVirtualProcess to work without having to provide an unwinder.

I don't think any of these scenarios are important or necessary for 2.1.

**Comment by BruceForstall** [comment]
@mikem8361 has anything changed here since the last update? Does this need to be fixed for 3.0?

**Comment by mikem8361** [comment]
I don't think this will happen for 3.0.  I don't have the time to learn the differences in the ELF arm unwind info to make this work.

**Comment by mikem8361** [comment]
For managed live debugging via the ICorDebug API, this isn't needed now for arm/arm64  now that the FEATURE_DATATARGET4 is defined.

Also, SOS under lldb works because it implements the DataTarget4's VirtualUnwind.

But as it turns out, implementing this for arm/arm64 is important for three other scenarios:

1) "createdump" uses PAL_VirtualUnwindOutOfProc directlry to make sure the code/stack for all the native stack frames.
2) dotnet-dump for linux/ELF core dumps on arm/arm64.  There is no way to implement the data target 4 interface.
3) Same with any CLRMD code using it's managed stack unwind API on Linux core dumps.

dotnet-dump/CLRMD will not be able to unwind the managed past any special HelperMethodFrames.

**Comment by mikem8361** [comment]
ARM64 is finished.  ARM32 still needs to be done.

**Comment by mikem8361** [comment]
This issue has been completed.

**Comment by noahfalk** [comment]
@mikem8361 - was this intended to be reopened? The last comment from you suggested it was done.

**Comment by mikem8361** [comment]
I think I left this open because the changes I made (in master) to the PAL's libunwind code for arm (and removed a bunch of code for x64/arm64) may have to be revisited. There are cases like source build for RHEL that require us to use the actual libunwind8 library and not the slightly changed copy we have in the PAL.

/cc: @janvorli
